docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+
+## Terminology
+
+The following terminology is used throughout this specification:
+
+| Name                 | References |
+| -------------------- | --- |
+| Waku node            | An Ethereum node with Waku V1 enabled, or a [10/WAKU2](https://rfc.vac.dev/spec/10/) node that implements [11/WAKU2-RELAY](https://rfc.vac.dev/spec/11/)|
+| Store node           | A Waku node that implements [11/WAKU-MAILSERVER](/spec/11) or [13/WAKU2-STORE](https://rfc.vac.dev/spec/13/) respectively |
+| Waku network         | A group of Waku nodes connected through the internet connection and forming a graph |
+| Community owner      | A Status user that owns a Status community |
+| Community member     | A Status user that is part of a Status community |
+| Community owner node | A Status node with message archive capabilities enabled, run by a community owner |
+| Community member node| A Status node with message archive capabilities enabled, run by a community member |
+| Live messages        | Waku messages received through the Waku network |
+
+
+## Supported platforms
+
+Creating, distributing, downloading and unpacking community history archives SHOULD only be supported in desktop clients. Mobile clients SHOULD NOT implement this functionality due to bandwidth and storage limitations.
+
+## Requirements / Assumptions
+
+This specification has the following assumptions:
+
+- Store nodes are available 24/7, ensuring constant live message availability
+- Store nodes have enough storage to persist historical messages for up to 30 days
+- No store nodes have storage to persist historical messages older than 30 days
+- All nodes are honest
+- The network is reliable
+
+Furthermore, it assumes that:
+
+- Community owner nodes have enough storage to persist historical messages older than 30 days
+- Community owner nodes provide archives with historical messages **at least** every 30 days
+- Community owner nodes receive all community messages
+- Community owner nodes are honest
+
+## Overview
+
+The following is a high-level overview of the user flow and features this specification describes.
+
+### Serving community history archives
+
+Community owner nodes go through the following (high level) process to provide community members with message histories (assumes community owner node is available 24/7):
+
+1. Community owner creates a Status community
+2. Community owner enables community history archive support
+3. A special type of channel for distributing magnet links ([Magnet URI scheme](https://en.wikipedia.org/wiki/Magnet_URI_scheme), [Extensions for Peers to Send Metadata Files](https://www.bittorrent.org/beps/bep_0009.html)) is created
+4. Community owner invites members and creates additional channels
+5. Community owner node receives messages and stores them into local database
+6. After 7 days, the community owner node exports and compresses last 7 days worth of messages from database and creates a magnet link from that data via torrent client
```

Comment:
@John-44 I think your initial proposal said the owner is gonna do this after 14 days initially and **then** do it subsequentially every 7 days. I wasn't sure why that was, so I want with 7 days right away. Happy to change this.

Reply:
@PascalPrecht I meant 7 days initially, sorry for not being clear on this.  So 7 days right away is correct :-)

Reply:
@PascalPrecht would it be worth updating this sentence to say:

"6. After 7 days, the community owner node exports and compresses last 7 days worth of messages from database *into a binary blob* and creates a magnet link from *the binary blob* via torrent client"?  

This change is to make it more explicit that the torrent operations are all dealing with a single binary blob that contains the compressed messages

Reply:
Generally I haven't explicitly said "blob" everywhere, because I think we can safely assume that computers send blobs at the end of the day (we also don't mention blobs in any other spec when talking about data).

> isn't the magnet link created in step 8?

As later in the document explicitly stated, there's actually multiple magnet links being created:

1. One for each archive (either every 7 days or for whatever time range messages were missed)
2. One for every time a new message archive index is published (which includes the magnet link to the previously created archive + magnet links for all archives prior to that. 

See "Bundling history archives into archive indices" section for a more detailed explaination.

Also, happy to add visuals to make this more clear.